src/containing_scope.py

```diff
@@ -0,0 +1,56 @@
+from talon import Context, Module
+
+mod = Module()
+ctx = Context()
+
+ctx.matches = r"""
+app: vscode
```

Comment:
Should we instead have a `cursorless` tag?  And then just activate it when the app is VSCode?

Reply:
Do these rely on the parsetree stuff or anything? A tag could potentially be useful to make the list "opt-in" as any necessary language-specific support is added.

Reply:
Oh that's interesting. I was thinking one big cursorless tag rather than using VSCode everywhere, but we could also use tags to turn certain aspects on and off

@knausj85 were you thinking a user would turn the tag entirely off for parsetree stuff or only based on language mode?  One thing to consider if per-language, is that then you wouldn't be able to refer to a syntactic element in opposite split if it's of a different language

Reply:
If you lose the ability to do the fancy split stuff, I don't think it's worth it. The thought was per-language.

```
app: vscode
and code.language: python
-
tag: user.cursorless_scope
```

Or something like that. `user.cursorless_scope` would be used for containing scope list, allowing us to effectively restrict the relevant commands to languages that support it.

...

As for the broader cursorless tag: I don't see any immediate advantage of tag vs app on the lists themselves.

A tag might make more sense to toggle `cursorless.talon`... if there are contexts within VSCode where we want to toggle the commands entirely.

Reply:
I went ahead and created a cursorless tag, but it's automatic activated by cursorless.talon when the app is vscode.
Just a provision for the future.

Reply:
That feels nicer to me. Because really what we're saying semantically is "this stuff should be active when cursorless is active". Today that happens to be whenever VSCode is the main app but that doesn't necessarily have to be the case in the long term. @knausj85 Any strong objections to that?